src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -281,21 +319,303 @@ bool UnixNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
+            trailingEpilogueInstructions++;
+        }
+        else if (IS_REX_PREFIX(NextByte[0]) && ((NextByte[1] & 0xf8) == POP_OP))
+        {
+            NextByte += 2;
+            trailingEpilogueInstructions++;
+        }
+        else
+        {
+            break;
+        }
+    }
+
+    //
+    // A REPNE prefix may optionally precede a control transfer
+    // instruction with no effect on unwinding.
+    //
+
+    if (NextByte[0] == REPNE_PREFIX)
+    {
+        NextByte += 1;
+    }
+
+    if (((NextByte[0] == RET_OP) ||
+        (NextByte[0] == RET_OP_2)) ||
+        (((NextByte[0] == REP_PREFIX) && (NextByte[1] == RET_OP))))
+    {
+        //
+        // A return is an unambiguous indication of an epilogue.
+        //
+        return trailingEpilogueInstructions;
+    }
+
+    if ((NextByte[0] == JMP_IMM8_OP) ||
+        (NextByte[0] == JMP_IMM32_OP))
+    {
+        //
+        // An unconditional branch to a target that is equal to the start of
+        // or outside of this routine is logically a call to another function.
+        //
+
+        size_t BranchTarget = (size_t)NextByte;
+        if (NextByte[0] == JMP_IMM8_OP)
+        {
+            BranchTarget += 2 + (int8_t)NextByte[1];
+        }
+        else
+        {
+            uint32_t delta =
+                (uint32_t)NextByte[1] |
+                ((uint32_t)NextByte[2] << 8) |
+                ((uint32_t)NextByte[3] << 16) |
+                ((uint32_t)NextByte[4] << 24);
+
+            BranchTarget += 5 + (int32_t)delta;
+        }
+
+        //
+        // Determine whether the branch target refers to code within this
+        // function. If not, then it is an epilogue indicator.
+        //
+        // A branch to the start of self implies a recursive call, so
+        // is treated as an epilogue.
+        //
+
+        size_t startAddress;
+        size_t endAddress;
+        uintptr_t lsda;
+
+        bool result = FindProcInfo((uintptr_t)pvAddress, &startAddress, &endAddress, &lsda);
+        ASSERT(result);
+
+        if (BranchTarget < startAddress || BranchTarget >= endAddress)
+        {
+            return trailingEpilogueInstructions;
+        }
+    }
+    else if ((NextByte[0] == JMP_IND_OP) && (NextByte[1] == 0x25))
+    {
+        //
+        // An unconditional jump indirect.
+        //
+        // This is a jmp outside of the function, probably a tail call
+        // to an import function.
+        //
+
+        return trailingEpilogueInstructions;
+    }
+    else if (((NextByte[0] & 0xf8) == SIZE64_PREFIX) &&
+        (NextByte[1] == 0xff) &&
+        (NextByte[2] & 0x38) == 0x20)
+    {
+        //
+        // This is an indirect jump opcode: 0x48 0xff /4.  The 64-bit
+        // flag (REX.W) is always redundant here, so its presence is
+        // overloaded to indicate a branch out of the function - a tail
+        // call.
+        //
+        // Such an opcode is an unambiguous epilogue indication.
+        //
+
```

Comment:
You may want to check for a breakpoint and give up if you see one. 

(It is annoying that setting a breakpoint introduces GC crash.)

Reply:
I think 'int 3' in regular code will be naturally handled since it is not recognized as a part of prologue anyways. 

Placing a breakpoint in epilogue might be rare and would be hard to handle reliably, since I would not know what it replaced. 
I will have to think about this. I can treat `int 3` as a `ret` if I have seen pops.

Reply:
Or just return -1 if i see `int 3`and treat that as not-hijackable.